geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/stats/MetadataBackedStats.scala

```diff
@@ -136,14 +140,41 @@ trait MetadataBackedStats extends GeoMesaStats with StatsBasedEstimator with Laz
     writeStat(new SeqStat(sft, stats), sft, merge = false)
 
     // update our last run time
-    val date = GeoToolsDateFormat.format(Instant.now().atZone(ZoneOffset.UTC))
-    ds.metadata.insert(sft.getTypeName, GeoMesaMetadata.STATS_GENERATION_KEY, date)
+    ds match {
+      case hm: HasGeoMesaMetadata[String] =>
+        val date = GeoToolsDateFormat.format(Instant.now().atZone(ZoneOffset.UTC))
+        hm.metadata.insert(sft.getTypeName, GeoMesaMetadata.STATS_GENERATION_KEY, date)
+      case _ => // no-op
+    }
 
     stats
   }
 
+  override def runStats[T <: Stat](sft: SimpleFeatureType, stats: String, filter: Filter): Seq[T] = {
```

Comment:
previously this was delegated to individual data stores, but at this point all data stores can handle a 'stats' type query